packages/lesswrong/components/posts/usePostsItem.ts

```diff
@@ -140,10 +140,10 @@ export const usePostsItem = ({
   );
 
   const compareVisitedAndCommentedAt = (
-    lastVisitedAt: Date,
+    lastVisitedAt: Date | null,
     lastCommentedAt: Date | null,
   ) => {
-    const newComments = lastCommentedAt ? lastVisitedAt < lastCommentedAt : false;
+    const newComments = lastCommentedAt ? lastVisitedAt ? lastVisitedAt < lastCommentedAt : true : false;
```

Comment:
No nested ternaries, plz.  I think this should be `(lastCommentedAt && lastVisitedAt) ? (lastVisitedAt < lastCommentedAt) : false`, but would double-check that logic.  (And frankly might just prefer to extract that out to a helper function that uses an actual if-statement for readability's sake.)

Reply:
Oh, see also https://github.com/ForumMagnum/ForumMagnum/pull/8686/files#r1464340258; in this case I'm less sure about the intended behavior and suspect changes are more likely to be mistakes.